lib/models/jellyfin_models.dart

```diff
@@ -2268,7 +2270,13 @@ class BaseItemDto with RunTimeTickDuration {
 
   factory BaseItemDto.fromJson(Map<String, dynamic> json) =>
       _$BaseItemDtoFromJson(json);
-  Map<String, dynamic> toJson() => _$BaseItemDtoToJson(this);
+  Map<String, dynamic> toJson({bool setOffline = true}) {
+    var json = _$BaseItemDtoToJson(this);
+    if (setOffline) {
+      json["FinampOffline"] = true;
+    }
+    return json;
+  }
```

Comment:
Is lacing the `BaseItemDto` with Finamp state information a good idea? Sure we have custom getters, but introducing a custom property which is also included in the JSON (and therefore potentially sent to the server at some point) seems drastic.

Reply:
As far as I can tell, the property is used to mark items where we prefer the favorite data from the server? Maybe tracking this in a separate list or map would be better, especially since lookup times don't appear to be critical here as it happens in the background anyway?

Reply:
Yeah, it's a bit hacky, but I couldn't figure out any other reasonable way to do this.  Note that it isn't id associated - that is, we might have two BaseItemDtos with identical IDs for the same item, but one should be trusted and one should not be.  This makes any sort of separate lookup difficult.  The attribute needs to be included in the JSON to survive being serialized by DownloadItems and FinampQueueItems.  As of right now we never send a BaseItemDto to the server, and I don't really see why we ever would, only ids or other specific attributes, so I'm hoping that never comes up as an issue.

Reply:
I'm not sure if editing metadata would require sending BaseItemDtos. If not, it should be fine.

But what if we add the needed metadata to the DownloadItem and MediaItem directly? Or is that too complicated/convoluted?

Reply:
It gets messy and convoluted fast, because we pass BaseItemDtos around all over the place.  Tracking their origin externally would be pretty hard.  If we somehow end up sending BaseItemDtos to the server, we can just make sure to strip this field out first.

Reply:
Okay, let's leave it like that for now.  
FYI there are also plans to migrate Finamp to using https://github.com/TentacleOrg/Tentacle (which is maintained by Kara from JellyBook, and meant as a collaboration between [JellyBook](https://github.com/JellyBookOrg/JellyBook), [Jellyflix](https://github.com/jellyflix-app/jellyflix) and Finamp) instead of our current manually-maintained API client. For that we will probably need to find a compromise between adding additional features (like custom getters and setters) to tentacle, and moving some of our custom stuff (like this BaseItemDto modification) to another abstraction layer in Finamp.